exonum-java-binding/core/rust/integration_tests/tests/service_runtime.rs

```diff
@@ -65,54 +65,80 @@ fn load_two_services() {
 }
 
 #[test]
-#[should_panic(expected = "Unable to load artifact")]
 fn load_nonexistent_artifact() {
     let runtime = get_runtime();
     let artifact_path = "nonexistent_artifact.jar";
-    runtime.load_artifact(&artifact_path);
+
+    assert_panics(
+        &format!("Unable to load artifact {}", artifact_path),
+        || runtime.load_artifact(&artifact_path),
+    );
 }
 
 #[test]
-#[should_panic(expected = "Unable to load artifact")]
 fn load_artifact_twice() {
-    let res = catch_unwind(|| {
-        let runtime = get_runtime();
-        let artifact_path = create_service_artifact_valid(runtime.get_executor());
-        runtime.load_artifact(&artifact_path);
-        (runtime, artifact_path)
-    });
-    assert!(res.is_ok());
+    let runtime = get_runtime();
+    let artifact_path = create_service_artifact_valid(runtime.get_executor());
+    runtime.load_artifact(&artifact_path);
 
-    let (runtime, artifact_path) = res.unwrap();
     // The second loading attempt should fail.
-    runtime.load_artifact(&artifact_path);
+    assert_panics(
+        &format!(
+            "Unable to load artifact {}",
+            artifact_path.to_string_lossy()
+        ),
+        || runtime.load_artifact(&artifact_path),
+    );
 }
 
 #[test]
-#[should_panic(expected = "Unable to load artifact")]
 fn load_failing_artifact() {
     let runtime = get_runtime();
     let artifact_path = create_service_artifact_non_loadable(runtime.get_executor());
-    runtime.load_artifact(&artifact_path);
+
+    assert_panics("Unable to load artifact", || {
+        runtime.load_artifact(&artifact_path)
+    });
 }
 
 #[test]
-#[should_panic(expected = "Unable to create service for artifact_id")]
 fn non_instantiable_service() {
     let runtime = get_runtime();
     let artifact_path = create_service_artifact_non_instantiable_service(runtime.get_executor());
     let artifact_id = runtime.load_artifact(&artifact_path);
-    runtime.create_service(&artifact_id);
+
+    assert_panics(
+        &format!("Unable to create service for artifact_id [{}]", artifact_id),
+        || runtime.create_service(&artifact_id),
+    );
 }
 
 #[test]
-#[should_panic(expected = "Unable to create service for artifact_id")]
 fn create_service_for_unknown_artifact() {
     let runtime = get_runtime();
-    runtime.create_service("unknown:artifact:id");
+
+    assert_panics(
+        "Unable to create service for artifact_id [unknown:artifact:id]",
+        || runtime.create_service("unknown:artifact:id"),
+    );
 }
 
 // Creates a new instance of JavaServiceRuntime for same JVM.
 fn get_runtime() -> JavaServiceRuntime {
     JavaServiceRuntime::create_with_jvm(VM.clone(), 0)
 }
+
+// Asserts that given closure panics while executed.
+fn assert_panics<F, R>(err_pattern: &str, f: F)
```

Comment:
Is it a pattern or, as the code below suggests, a sub-string?

Reply:
Actually, it's a substring, but it's called "err_pattern" because in fact it gets passed to the underlying functions that expect a `Pattern` entity and treated as a pattern inside a string. Do you want me to rename it to smth like `err_substring`?

Reply:
I am not sure I understand: can a user pass a pattern (a glob, a regex?), or only a substring?

Reply:
Ah, no, substring only, no patterns in term of regex.

Reply:
Then please name accordingly